feature/experimental/gribi/ate_tests/route_addition_during_failover_test/route_addition_during_failover_test.go

```diff
@@ -0,0 +1,668 @@
+	// SINGLE_PRIMARY mode, with FIB ACK requested. Specify gRIBI as the leader.
+	client := fluent.NewClient()
+	client.Connection().WithStub(gribic).WithPersistence().WithInitialElectionID(1, 0).
+		WithFIBACK().WithRedundancyMode(fluent.ElectedPrimaryClient)
+	client.Start(ctx, t)
+	defer client.Stop(t)
+
+	defer func() {
+		// Flush all entries after test.
+		if err := gribi.FlushAll(client); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	client.StartSending(ctx, t)
+	if err := awaitTimeout(ctx, client, t, time.Minute); err != nil {
+		t.Fatalf("Await got error during session negotiation for clientA: %v", err)
+	}
+	eID := gribi.BecomeLeader(t, client)
+
+	t.Log("Flush existing gRIBI routes before test.")
+	if err := gribi.FlushAll(client); err != nil {
+		t.Fatal(err)
+	}
+
+	args := &testArgs{
+		ctx:        ctx,
+		client:     client,
+		dut:        dut,
+		ate:        ate,
+		top:        top,
+		electionID: eID,
+	}
+
+	virtualIPs := createIPv4Entries(t, ipBlock1FlowArgs.ipBlock)
+
+	t.Log("Inject routes from ipBlock1 in default VRF with NHGID: #1.")
+	pushDefaultEntries(t, args, subIntfIPs, virtualIPs, configNhg, !switchover)
+
+	gr, err := args.client.Get().
+		WithNetworkInstance(*deviations.DefaultNetworkInstance).
+		WithAFT(fluent.IPv4).
+		Send()
+	if err != nil {
+		t.Fatalf("Error encountered during gRIBI Get operation: got: %v", err)
+	}
+
+	for _, r := range gr.GetEntry() {
+		entry := r.GetIpv4().Prefix
+		if got := r.GetFibStatus().String(); got != "PROGRAMMED" {
+			t.Fatalf("gRIBI entry %v is not programmed in FIB, got:%v, want: PROGRAMMED", entry, got)
+		}
+	}
+
+	// Send traffic from ATE port-1 to prefixes in ipBlock1 and ensure traffic
+	// flows 100% and reaches ATE port-2.
+	testTraffic(t, *args, createTrafficFlow(t, args.ate, args.top, ipBlock1FlowArgs))
+
+	controllers := cmp.FindComponentsByType(t, dut, controllerCardType)
+	t.Logf("Found controller list: %v", controllers)
+	// Only perform the switchover for the chassis with dual controllers.
+	if len(controllers) != 2 {
+		t.Skipf("Dual controllers required on %v: got %v, want 2", dut.Model(), len(controllers))
+	}
+
+	secondaryBeforeSwitch, primaryBeforeSwitch := findController(t, dut, controllers)
+
+	awaitSwitchoverReady(t, dut, primaryBeforeSwitch)
+	t.Logf("Controller %q is ready for switchover before test.", primaryBeforeSwitch)
+
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	switchoverRequest := &spb.SwitchControlProcessorRequest{
+		ControlProcessor: cmp.GetSubcomponentPath(secondaryBeforeSwitch),
+	}
+	t.Logf("switchoverRequest: %v", switchoverRequest)
+
+	// Concurrently run switchover and gribi route addition in ipBlock2.
+	virtualIPsBlock2 := createIPv4Entries(t, ipBlock2FlowArgs.ipBlock)
+
+	// Check for coredumps in the DUT and validate that none are present on DUT before switchover.
+	coreFileCheck(t, dut, gnoiClient, sysConfigTime)
+
+	t.Log("Execute gRIBi route addition and master switchover concurrently.")
+	go func(msg string) {
```

Comment:
The msg parameter is unused, and why put this in a goroutine? I don't see any logic that waits for the routine to be done.

Reply:
removed it, thank you